project-koku/koku
# COST-3043: GCP instance-type with group_by/filter shows all data

## Jira Ticket

[COST-3043](https://issues.redhat.com/browse/COST-3043)

## Description

When hitting the default table during a group_by or filter option on the instance type endpoint, we were missing the 
`sku_alias LIKE '%%Instance Core running%%'` filter we use to build our UI tables. 

## Testing
I tested this using smoke tests locally with: 
1. pull the latest iqe plugin and make sure you use nise >= 3.0:

2. make sure we have no "Instance Core running" data and we still have some Compute data :
    - in iqe_cost_management/data/gcp/gcp_static_report.yml: replace `sku_id: CF4E-A0C7-E3BF` with  `sku_id: 8C61-80B1-C43A`
    
3. put `breakpoint()` into test_api_gcp_ingest_single_source and execute the test
 `ENV_FOR_DYNACONF=local iqe tests plugin cost_management -k "test_api_gcp_ingest_single" -vv --pdb`

3. hit http://localhost:8000/api/cost-management/v1/reports/gcp/instance-types/

you will see no data (expected)

4. hit

http://localhost:8000/api/cost-management/v1/reports/gcp/instance-types/?group_by[service]=*

- On main you will see data
- On this branch you will not see data (remember to clear out your cache inbetween with make clear-cache)
